### DISCUSSION TITLE
More intuitive/fluid controls (e.g. drag-n-drop, right-click user rename)

### DISCUSSION THREAD

**Discussion by goodnightmonkey** [stoatchat/discussions#563] [Feature Suggestions]
Revolt is generally very good! I could figure out how to access or execute a variety of things, but there were some basic controls(?) that I found were either very clunky/inconvenient, or not intuitive. 
Also, I know this sounds like multiple suggestions, but it's really just "more intuitive/fluid" controls as a whole. These are just some examples of what I mean by more intuitive/fluid controls.

(TL;DR: right-click user menu from user icons/names right in the server + easy re-arrange/add categories and channels)

- Adding, removing, re-arranging server channels & categories
     - Right now, you can right-click add categories and channels on servers from the server itself, but you can't re-order channels from within the server's sidebar, along with other basic functions. I had to go to Server settings and do everything that way; which makes sense so one can't accidentally rearrange while viewing the server, but it wasn't very evident and it was kind of a pain (I also kept accidentally deleting categories). I think I'd like there to be either both options where one can edit via sidebar or settings - or - there just be the editing from the sidebar option. If people are concerned about accidentally rearranging channels and categories, there could be a locking feature that prevents them from being rearranged until unlocked (kind of like locking layers in Photoshop).
  
- Right-clicking your own name/icon in a server to change server name and/or server icon etc
     -  I think right clicking on the server's icon is fine and makes sense for super large and active communities, but not intuitive. I would personally like to be able to just right click on my icon/username from the chat itself, and the member list. I think it's more intuitive to have member-related menus be acquired by clicking on the members themselves.
 
- Admin changing another user's server display name (not their actual display or username)
     - So, sometimes in my experience of running a server, some people will use inappropriate display names or break some other server rule with their display name, or we'll have a holiday event where everyone makes holiday puns with their names but some people forget to change their server display names or they can't figure out how. I find it (usually) easier to just change their server display name  and let them know. I haven't seen that option, so I think it'd be great addition for extra security for server admins.
     
- General right-click update user for admins (e.g. time out, kick, change name, add role, etc)
     - I'm not sure if this is because I'm testing in a 1 person server right now, but I can't find where to do basic staff tasks, like timout, kick, change name, and so on. If those features don't already exist, could a user menu be added just by right clicking on the user?

Thanks for reading.

**Comment by Rexogamer** [comment]
Regarding your third point, the API does let admins edit other member's nicknames but this isn't exposed in the app itself yet - however, you can use AutoMod's /nick command for this. In addition, you *should* be able to right click on other users (certainly in the member list) to kick, ban and timeout them - this might not be visible if your testing server only contains you, as for obvious reasons you can't kick/ban/timeout yourself
